app/collections/create-form.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import Translate from 'react-translate-component';
+import apiClient from 'panoptes-client/lib/api-client';
+
+class CollectionsCreateForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      collectionNameLength: 0,
+      error: null
+    };
+
+    this.onSubmit = this.onSubmit.bind(this);
+    this.handleNameInputChange = this.handleNameInputChange.bind(this);
+  }
+
+  onSubmit(event) {
+    event.preventDefault();
+
+    const displayName = this.name.value;
+    const isPrivate = this.isPrivate.checked;
+
+    const links = {};
+
+    if (this.props.project) {
+      links.project = this.props.project.links;
+    }
+
+    if (this.props.subjectIDs.length > 0) {
+      links.subjects = this.props.subjectIDs;
+    }
+
+    const collection = {
+      display_name: displayName,
+      private: isPrivate,
+      links
+    };
+
+    apiClient.type('collections').create(collection).save()
+      .then((newCollection) => {
+        this.name.value = '';
+        this.isPrivate.value = true;
+        this.props.onSubmit(newCollection);
+      })
+      .catch((error) => {
+        this.setState({ error });
+      });
+  }
+
+  handleNameInputChange() {
+    this.setState({
+      collectionNameLength: this.name.value.length
+    });
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.onSubmit} className="collections-create-form">
+        <div className="form-help error">
+          {this.state.error &&
+              `Error ${this.state.error.status}: ${this.state.error.message}`}
```

Comment:
The conversion here is missing the more user friendly messages. We should still use those. Now that we have the status codes available to us, that could be useful in determining what message should be.